docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -0,0 +1,186 @@
+  It can have parameters which can be used to filter a subset of the Workflow's state that it returns. Since it does return a value it must have a non `void` return type.
+
+Workflow interfaces can define only a single method annotated with `@WorkflowMethod`. They can define
+any number of methods annotated with `@SignalMethod` and `@QueryMethod`, for example:
+
+```java
+@WorkflowInterface
+public interface FileProcessingWorkflow {
+
+    @WorkflowMethod
+    String processFile(Arguments args);
+
+    @QueryMethod(name="history")
+    List<String> getHistory();
+
+    @QueryMethod
+    String getStatus();
+
+    @SignalMethod
+    void retryNow();
+
+    @SignalMethod
+    void abandon();
+}
+```
+
+The `@WorkflowMethod` annotation has a `name` parameter, for example: `@WorkflowMethod(name = "MyWorkflowType")`.
+It can be used to denote the Workflow type. If not set, the Workflow type defaults to the short name of the Workflow interface. In the example above, the Workflow type defaults to `FileProcessingWorkflow`.
+Methods annotated with `@WorkflowMethod` can have any number of parameters. We recommend passing a single parameter that contains all the input fields. This allows adding fields in a backward compatible manner.
+
+The `@QueryMethod` annotation also has a `name` parameter, for example: `@QueryMethod(name = "history")`. It can be
+used to denote the Query name. If not set, the Query name defaults to the name of the method. In the example above, the Query name defaults to `getStatus`.
+
+The `@SignalMethod` too has a `name` parameter, for example: `@SignalMethod(name = "mysignal")`. It can be used to denote the
+Signal type. If not set, the Signal type defaults to the name of the method. In the example above, the Signal type defaults to `retryNow` and `abandon`.
+
+### Workflow Interface Inheritance
+
+Workflow interfaces can form inheritance hierarchies. It may be useful for creating components reusable across multiple
+Workflow interfaces. For example, to implement a UI or CLI button that allows you to call `retryNow` Signal on any Workflow, redesign the above interface example to:
+
+```java
+public interface Retryable {
+    @SignalMethod
+    void retryNow();
+}
+
+@WorkflowInterface
+public interface FileProcessingWorkflow extends Retryable {
+
+    @WorkflowMethod
+    String processFile(Arguments args);
+
+    @QueryMethod(name="history")
+    List<String> getHistory();
+
+    @QueryMethod
+    String getStatus();
+
+    @SignalMethod
+    void abandon();
+}
+```
+
+With this, another Workflow interface can extend just `Retryable`, for example:
+
+```java
+@WorkflowInterface
+public interface MediaProcessingWorkflow extends Retryable {
+
+    @WorkflowMethod
+    String processBlob(Arguments args);
+}
+```
+
+Now if we have two running Workflows, one that implements the `FileProcessingWorkflow` interface and another that implements the
```

Comment:
This content maps to "How to send a Signal to a Workflow Execution in Java" I think.